src/srcset.js

```diff
@@ -197,10 +197,10 @@ export class Srcset {
    * @private
    */
   selectByWidth_(width, dpr) {
-    let minIndex = -1;
-    let minWidth = 1000000;
-    let minScore = 1000000;
-    for (let i = 0; i < this.sources_.length; i++) {
+    const length = this.sources_.length;
+    const minWidth = this.sources_[length - 1].width / dpr;
```

Comment:
Actually, just realized that `width: undefined` and `width: 0` cases are already disallowed. So, that branch is meaningless. Removed. Thanks for the catch!